### PR TITLE
Add GitDependency support to git repositories as extdep.

### PR DESCRIPTION
**Pull request by spbrogan**
Minor changes needed to RepoResolver to expose the Repo Object.

### CODE REVIEW

File: MuEnvironment/GitDependency.py [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+            # valid repo folder
+            r = Repo(self._local_repo_root_path)
+            if(not r.initalized):
+                self.logger.error("Git Dependency: Not Initialized")
+                result = False
+            elif(r.dirty):
+                self.logger.error("Git Dependency: dirty")
+                result = False
+
+            if(r.head.commit != self.version):
```
Comment (matthewfcarlson) [inline MuEnvironment/GitDependency.py:96]: So we only support tracking a specific commit, do we plan on tracking a specific branch?

File: MuEnvironment/tests/test_GitDependency.py [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+uptodate_version = "7fd1a60b01f91b314f59955a4e4d4e80d8edf11d"
+behind_one_version = "762941318ee16e59dabbacb1b4049eec22f0d303"
+invalid_version = "762941318ee16e59d123456789049eec22f0d303"
+
+hw_json_template = '''
+{
+  "scope": "global",
+  "type": "git",
+  "name": "HelloWorld",
+  "source": "https://github.com/octocat/Hello-World.git",
```
Comment (matthewfcarlson) [inline MuEnvironment/tests/test_GitDependency.py:47]: Good choice for a repo

File: README.rst [hunk 3]
```diff
@@ -19,16 +19,23 @@ Please see Project Mu for details https://microsoft.github.io/mu
 Version History
 ===============
 
+0.3..10.Next
```
Comment (matthewfcarlson) [inline README.rst]: We typically do 0.3.11-dev, but I'm fine with this

**Review by matthewfcarlson — APPROVED** [review]

File: MuEnvironment/GitDependency.py (hunk 1 above)
Comment (spbrogan) [reply MuEnvironment/GitDependency.py:96]: Ext_dep fit better with just a commit as version.  I think it should be put on the list to be added later

**Pull request merged**